### PR TITLE
nfc: Add Launch App support

### PR DESCRIPTION
**Pull request by dchat-nordic**
Added:
-Ported Launch App library from NRF5 SDK
-Added docs
-Added Launch App sample
-Added Universal Link support (for iOS)

Signed-off-by: Dominik Chat <dominik.chat@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -137,14 +152,6 @@ nRF9160 samples
 
   * Modified to use runtime location service selection instead of compile-time configurations.
 
-Matter samples
```
Comment (grochu) [inline doc/nrf/releases/release-notes-changelog.rst:140]: Is this a side effect of resolving a rebase conflict?

File: samples/nfc/record_launch_app/README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+After programming the sample to your development kit, perform the following steps to test it:
+
+1. Touch the NFC antenna with the smartphone or tablet and observe that **LED 1** is lit.
+#. Observe that the smartphone or tablet launches the nRF Toolbox application.
```
Comment (grochu) [inline samples/nfc/record_launch_app/README.rst:53]: Shall we clarify the smartphone either launches the application or opens an appstore with the app?

**Review by grochu — COMMENTED** [review]
Looks very good. Just added a few minor comments.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (dchat-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:140]: Yes, the entries should be ordered alphabetically, so I moved matter into correct place.

File: samples/nfc/record_launch_app/README.rst (hunk 2 above)
Comment (dchat-nordic) [reply samples/nfc/record_launch_app/README.rst:53]: I will add this information now.

**Review by grochu — APPROVED** [review]
Looks good

File: samples/nfc/record_launch_app/README.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#. Move the smartphone or tablet away from the NFC antenna and observe that **LED 1**
+   turns off.
+
+   .. note::
+	   Devices running iOS require the nRF Toolbox app to be installed before testing the sample.
+	   Devices running Android will open Google Play when application is not installed.
```
Comment (peknis) [inline samples/nfc/record_launch_app/README.rst:59]:
```suggestion
	   Devices running Android will open Google Play when the application is not installed.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/nfc/record_launch_app/README.rst (hunk 3 above)
Comment (dchat-nordic) [reply samples/nfc/record_launch_app/README.rst]: Ok, added the change

**Review by peknis — APPROVED** [review]

**Pull request merged**